server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1127,8 +1127,10 @@ private Map<String, String> extractApiParamAsMap(Map ds) {
         return details;
     }
 
+    @Override
     @ActionEvent(eventType = EventTypes.EVENT_DISABLE_PRIMARY_STORAGE, eventDescription = "disable storage pool")
-    private void disablePrimaryStoragePool(StoragePoolVO primaryStorage) {
+    public StoragePool disablePrimaryStoragePool(Long id) {
+        StoragePoolVO primaryStorage = _storagePoolDao.findById(id);
         if (!primaryStorage.getStatus().equals(StoragePoolStatus.Up)) {
```

Comment:
@harikrishna-patnala might result in NPE on enable/disable for removed pools, it is checked in update storage pool flow earlier.

Reply:
added the ID check @sureshanaparti